File: CS/WindowsApplication3/Main.cs
using DevExpress.XtraEditors;
using DevExpress.XtraGrid;
using System;
using System.Data;
using System.Windows.Forms;


namespace DXSample
{
    public partial class Main : XtraForm
    {
        public Main()
        {
            InitializeComponent();
        }

        DataTable GetCustomerInfoDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "CustomerInfo";
            table.Columns.Add(new DataColumn("CustomerID", typeof(int)));
            table.Columns.Add(new DataColumn("FirstName", typeof(string)));
            table.Columns.Add(new DataColumn("LastName", typeof(string)));
            table.Columns.Add(new DataColumn("Date", typeof(DateTime)));
            table.Columns.Add(new DataColumn("Check", typeof(bool)));
            for(int i = 0; i < 5; i++)
            {
                table.Rows.Add(i, string.Format("FirstName {0}", i), i, DateTime.Today.AddDays(i), i % 2 == 0);
            }
            return table;
        }

        private void OnEditValueChanged(object sender, EventArgs e)
        {
            if(gridView.FocusedRowHandle == GridControl.NewItemRowHandle)
            {
                gridView.CloseEditor();
                gridView.UpdateCurrentRow();
                gridView.ShowEditor();
                TextEdit edit = sender as TextEdit;
                if(edit == null) return;
                edit.SelectionStart = 1;
                edit.SelectionLength = 0;
            }
        }
        private void OnLoad(object sender, EventArgs e)
        {
            InitData();
            gridControl.ForceInitialize();
            for(int i = 0; i < gridView.VisibleColumns.Count; i++)
                gridView.VisibleColumns[i].RealColumnEdit.EditValueChanged += OnEditValueChanged;
        }

        protected override void OnFormClosing(FormClosingEventArgs e)
        {
            for(int i = 0; i < gridView.VisibleColumns.Count; i++)
                gridView.VisibleColumns[i].RealColumnEdit.EditValueChanged -= OnEditValueChanged;
            base.OnFormClosing(e);
        }

        public void InitData()
        {
            customerInfoBindingSource.DataSource = GetCustomerInfoDataTable();
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid
Imports System
Imports System.Data
Imports System.Windows.Forms

Namespace DXSample
    Partial Public Class Main
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Function GetCustomerInfoDataTable() As DataTable
            Dim table As DataTable = New DataTable()
            table.TableName = "CustomerInfo"
            table.Columns.Add(New DataColumn("CustomerID", GetType(Integer)))
            table.Columns.Add(New DataColumn("FirstName", GetType(String)))
            table.Columns.Add(New DataColumn("LastName", GetType(String)))
            table.Columns.Add(New DataColumn("Date", GetType(DateTime)))
            table.Columns.Add(New DataColumn("Check", GetType(Boolean)))
            For i As Integer = 0 To 5 - 1
                table.Rows.Add(i, String.Format("FirstName {0}", i), i, DateTime.Today.AddDays(i), i Mod 2 = 0)
            Next
            Return table
        End Function

        Private Sub OnEditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            If gridView.FocusedRowHandle = GridControl.NewItemRowHandle Then
                gridView.CloseEditor()
                gridView.UpdateCurrentRow()
                gridView.ShowEditor()
                Dim edit As TextEdit = TryCast(sender, TextEdit)
                If edit Is Nothing Then Return
                edit.SelectionStart = 1
                edit.SelectionLength = 0
            End If
        End Sub

        Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            InitData()
            gridControl.ForceInitialize()
            For i As Integer = 0 To gridView.VisibleColumns.Count - 1
                AddHandler gridView.VisibleColumns(i).RealColumnEdit.EditValueChanged, AddressOf OnEditValueChanged
            Next
        End Sub

        Protected Overrides Sub OnFormClosing(ByVal e As FormClosingEventArgs)
            For i As Integer = 0 To gridView.VisibleColumns.Count - 1
                AddHandler gridView.VisibleColumns(i).RealColumnEdit.EditValueChanged, AddressOf OnEditValueChanged
            Next
            MyBase.OnFormClosing(e)
        End Sub

        Public Sub InitData()
            customerInfoBindingSource.DataSource = GetCustomerInfoDataTable()
        End Sub
    End Class
End Namespace
